dwc_server.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import tensorflow as tf
+import os
+#import asyncio
+from fastapi import FastAPI, File, UploadFile
+from fastapi.responses import FileResponse
+#import websockets
+import copy
+import numpy as np
+import uuid
+
+app = FastAPI()
+
+datasets = []
+num_clients = 1
+model = None
+priors = []
+client_ids = []
+
+@app.get("/")
+async def root():
+    return "root"
+
+@app.get("/{model}/")
+async def update_model(model: tf.keras.Model):
+    return FileResponse('consolidated.h5')
+
+@app.post("/weights")
+async def load_weights(image: UploadFile = File(...)): #do we still need to load a file here
+    model.load_weights(image.filename)
+    datasets.append(copy.deepcopy(model))
+    dwc_frame()
+    return {'consolidated weights': model}
+
+def load_data():
+    datasets = []
+    client_ids = []
+    path = os.getcwd()
+    for root, dirs, files in os.walk(path):
+        for filename in files:
+            if filename[:6] != 'model-':
+                continue
+            model.load_weights(filename)
+            datasets.append(copy.deepcopy(model))
+            client_ids.append(filename[6:-3])
+
+def load_priors():
+    if not priors:
+        prior = None
+        for client in client_ids:
+            filename = 'prior-'+client+'.h5'
+            prior.load_weights(filename)
+            priors.append(copy.deepcopy(prior))
+
+async def dwc_frame():
+    load_data()
+    load_priors()
+    
+    model = distributed_weight_consolidation(datasets, priors)
+    
+    model.save_weights(
+        'server/consolidated-'+uuid.uuid4().__str__()+'.h5',
+        save_format = 'h5')
+    
+    update_model(model)
```

Comment:
this doesn't need to be called here.

Reply:
just the `update_model(model)`? or the save model as well?